clang/lib/Driver/Driver.cpp

```diff
@@ -5567,6 +5567,13 @@ void Driver::BuildActions(Compilation &C, DerivedArgList &Args,
 
   handleArguments(C, Args, Inputs, Actions);
 
+  auto *OptO = C.getArgs().getLastArg(options::OPT_o);
+  SmallString<128> OutFileDir(OptO ? OptO->getValues()[0] : "");
+  llvm::sys::path::remove_filename(OutFileDir);
+  // `remove_filename` will return an empty string if in CWD, patch it.
+  if (OutFileDir.empty())
+    llvm::sys::path::native(OutFileDir = "./");
```

Comment:
It is my understanding that we should use the `-o` information for the saved temporary files when `-save-temps=obj` is used.  Otherwise, the `CWD` is used.

Also, since `OutFileDir` is only used with `IsSaveTemps`, we can move this logic there.

Reply:
Agreed.
I kept the declaration of `OutFileDir` above the loop, so it doesn't need to be re-calculated.